test/Microsoft.ML.TestFramework/DataPipe/TestDataPipe.cs

```diff
@@ -17,6 +17,130 @@
 
 namespace Microsoft.ML.Runtime.RunTests
 {
+    public sealed partial class TestDataPipe : TestDataPipeBase
+    {
+        private static Float[] dataFloat = new Float[] { -0.0f, 0,  1, -1,  2, -2, Single.NaN, Single.MinValue,
+                Single.MaxValue, Single.Epsilon, Single.NegativeInfinity, Single.PositiveInfinity };
+        private static uint[] resultsFloat = new uint[] { 21, 21, 16, 16, 31, 17, 0, 23, 24, 15, 10, 7 };
+        private static uint[] resultsFloatSparse = new uint[] { 21, 21, 16, 21, 21 };
+        private static Double[] dataDouble = new Double[]   { -0.0, 0, 1, -1,  2, -2, Double.NaN, Double.MinValue,
+                Double.MaxValue, Double.Epsilon, Double.NegativeInfinity, Double.PositiveInfinity };
+        private static uint[] resultsDouble = new uint[] { 21, 21, 31, 17, 10, 15, 0, 16, 21, 15, 6, 30 };
+        private static uint[] resultsDoubleSparse = new uint[] { 21, 21, 31, 21, 21 };
+
+        [Fact]
+        public void TestHashTransformFloat()
+        {
+            TestHashTransformHelper(dataFloat, resultsFloat, NumberType.R4);
+        }
+
+        [Fact]
+        public void TestHashTransformFloatVector()
+        {
+            var data = new[] { dataFloat };
+            var results = new[] { resultsFloat };
+            TestHashTransformVectorHelper(data, results, NumberType.R4);
+        }
+
+        [Fact]
+        public void TestHashTransformFloatSparseVector()
+        {
+            var data = new VBuffer<Single>(5, 3, new float[] { -0.0f, 0, 1 }, new[] { 0, 1, 2 });
```

Comment:
>{ 0, 1, 2 } [](start = 84, length = 11)

Given that much of the complexity surrounding the underlying code is ensuring that you're mapping to the correct values, I might appreciate changing the test a little to a non-trivial index set... say, 0, 3, 4 or something.